### PR TITLE
fix: Fix compiler errors in catalog and mito crates

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Fix compiler errors in catalog and mito crates.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #555

### CODE REVIEW

**Comment by waynexia** [comment]
You need to update this branch since #738 is merged. Maybe checkout a new branch and cherry-pick your commits to make it cleaner.

File: src/mito/src/engine.rs [hunk 1]
```diff
@@ -519,13 +519,12 @@ impl<S: StorageEngine> MitoEngineInner<S> {
...
-    use common_query::physical_plan::RuntimeEnv;
+    use common_query::physical_plan::SessionContext;
     use common_recordbatch::util;
-    use datafusion_common::field_util::{FieldExt, SchemaExt};
-    use datatypes::prelude::{ConcreteDataType, ScalarVector};
+    use datatypes::prelude::ConcreteDataType;
     use datatypes::schema::{ColumnDefaultConstraint, ColumnSchema, SchemaBuilder};
     use datatypes::value::Value;
-    use datatypes::vectors::*;
+    use datatypes::vectors::{TimestampMillisecondVector, *};
```
Comment (v0y4g3r) [inline src/mito/src/engine.rs:527]:
```sugesstion
    use datatypes::vectors::*;
```

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/mito/src/engine.rs (hunk 1 above)
Comment (evenyag) [reply src/mito/src/engine.rs:527]: I would change to use specific vectors.

**Pull request merged**